app/lib/services/shift_notifications/update.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Services
+  module ShiftNotifications
+    class Update
+      include Procto.call
+      include Concord.new(:shift, :url, :current_user)
+      include Adamantium::Flat
+
+      delegate :need, :duration, :user, :user_id_was, to: :shift
+
+      def call
+        notification_hash[:users].each do |user|
+          SendTextMessageWorker.perform_async(user.phone, [notification_hash[:message], url].join(' '))
+        end
+      end
+
+      private
+
+      def notification_hash
+        if current_user == user && user
```

Comment:
I would consider using a case statement here.

Reply:
Note to self: don't leave feedback in conversations that were already marked as resolved :-)